src/screens/TopicPageScreen/index.js

```diff
@@ -142,20 +145,26 @@ const TopicPageScreen = (props) => {
     }
   };
 
+  const handleFollowData = (type) => {
+    if (type === 'sign') {
+      setFollowType('signed');
+      setIsFollow(true);
+    } else if (type === 'anon') {
+      setFollowType('incognito');
+      setIsFollow(true);
+    } else {
+      setFollowType('');
+      setIsFollow(false);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleFollowData` function has been introduced to manage the `followType` and `isFollow` states based on the input type. This seems like a good abstraction to handle follow data.
```diff
+ const handleFollowData = (type) => {
+   if (type === 'sign') {
+     setFollowType('signed');
+     setIsFollow(true);
+   } else if (type === 'anon') {
+     setFollowType('incognito');
+     setIsFollow(true);
+   } else {
+     setFollowType('');
+     setIsFollow(false);
+   }
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->